### DISCUSSION TITLE
how to delete users in local DB to change authentication mode?

### DISCUSSION THREAD

**Discussion by hellt** [goharbor/harbor#16817] [Q&A] [answered]
Hi, I found out that in order to change local DB auth to LDAP one needs to delete users from DB (and there is [no way](https://github.com/goharbor/harbor/issues/16112) to do this via harbor portal).

Would anyone know a way to delete users from a DB?

**Comment by hellt** [comment, marked as answer]
unswering my own q

1. make a backup of your database (`cp -r /data/database /somebackuppath`)
2. run `psql` in a harbor_db container (`docker exec -it <cid> bash` and then `psql`)
3. within psql shell connect to registry db - `\c registry`
4. then drop all users with user_id >2 (1 is admin, 2 is anonymous) - `delete from harbor_user where user_id > 2;`

**Reply by stonezdj** [reply]
After Harbor 2.5.0, the admin user could delete users in users and after all users are removed from local DB. the auth mode could be changed.
